week4/trie/thalesmg/Main.hs

```diff
@@ -0,0 +1,56 @@
+import qualified Data.Map.Lazy as M
```

Comment:
When I looked up the documentation of Data.Map, it seemed like it didn't have much in it (https://hackage.haskell.org/package/containers-0.5.8.1/docs/Data-Map.html). In Data.Map.Lazy there were many methods defined and in Data.Map's documentation there was a link to the Lazy version, so I thought that was the right one.

Does Data.Map have all the operations defined in the Lazy one?